picketlink/src/main/java/org/wildfly/extension/picketlink/idm/model/IdentityConfigurationRemoveStepHandler.java

```diff
@@ -0,0 +1,92 @@
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceName;
+import org.wildfly.extension.picketlink.common.model.ModelElement;
+import org.wildfly.extension.picketlink.idm.service.PartitionManagerService;
+
+import java.util.Set;
+
+import static org.wildfly.extension.picketlink.logging.PicketLinkLogger.ROOT_LOGGER;
+
+/**
+ * @author Pedro Silva
+ */
+public class IdentityConfigurationRemoveStepHandler extends RestartParentResourceRemoveHandler {
+
+    static final IdentityConfigurationRemoveStepHandler INSTANCE = new IdentityConfigurationRemoveStepHandler();
+
+    private IdentityConfigurationRemoveStepHandler() {
+        super(ModelElement.PARTITION_MANAGER.getName());
+    }
+
+    @Override
+    protected void updateModel(OperationContext context, ModelNode operation) throws OperationFailedException {
+        PathAddress partitionManagerAddress = getParentAddress(context.getCurrentAddress());
+        Resource partitionManagerResource = context.readResourceFromRoot(partitionManagerAddress);
+        ModelNode originalParentModel = Resource.Tools.readModel(partitionManagerResource);
+
+        checkIfLastConfiguration(context);
+
+        super.updateModel(context, operation);
+
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                PartitionManagerRemoveHandler.INSTANCE
+                    .removeIdentityStoreServices(context, originalParentModel, partitionManagerAddress.getLastElement().getValue(), context.getCurrentAddressValue());
+
+                context.completeStep(OperationContext.ResultHandler.NOOP_RESULT_HANDLER);
+            }
+        }, OperationContext.Stage.RUNTIME);
+    }
+
+    @Override
+    protected void recreateParentService(OperationContext context, PathAddress parentAddress, ModelNode parentModel) throws OperationFailedException {
+        PartitionManagerAddHandler.INSTANCE.createPartitionManagerService(context, parentAddress.getLastElement()
+            .getValue(), parentModel, false);
+    }
+
+    @Override
+    protected ServiceName getParentServiceName(PathAddress parentAddress) {
+        return PartitionManagerService.createServiceName(parentAddress.getLastElement().getValue());
+    }
+
+    private void checkIfLastConfiguration(OperationContext context) throws OperationFailedException {
+        PathAddress partitionManagerAddress = getParentAddress(context.getCurrentAddress());
+        Resource partitionManagerResource = context.readResourceFromRoot(partitionManagerAddress);
+        Set<Resource.ResourceEntry> configurations = partitionManagerResource.getChildren(ModelElement.IDENTITY_CONFIGURATION.getName());
+
+        if (configurations.isEmpty() || configurations.size() == 1) {
+            throw ROOT_LOGGER.idmNoIdentityConfigurationProvided();
+        }
+    }
```

Comment:
You are right. I'm going to pass the Resource itself instead of the context and avoid a 2nd read of the model.